ros/src/waypoint_updater/speed_calculator.py

```diff
@@ -17,60 +17,43 @@ def __init__(self, target_speed, current_speed=0.0,
                               target_acceleration, current_accleration,
                               acceleration_limit, jerk_limit)
         else:
-            # No acceleration. Current speed will be returned regardless of distance.
-            self.distance_speed_lut = [current_speed]
-
-        # The list of distances represented in the LUT
-        self.distances = [i * self.delta_d for i in range(len(self.distance_speed_lut))]
+            # No acceleration. Current speed/acceleration will be returned regardless of distance.
+            self.distances = [0.0]
+            self.velocities = [current_speed]
+            self.accelerations = [current_accleration]
 
     def __accelerate(self, target_speed, current_speed,
                      target_acceleration, current_accleration,
                      acceleration_limit, jerk_limit):
         # Calculate the velocity at each time-step when applying acceleration and jerk according to their max limits.
         # Instead of continuously calculate when the jerk must change sign such that the acceleration can be reduced to
         # zero at target speed, the iteration is started from both ends to find the speed overlap in the middle.
-        v0 = current_speed
-        a0 = current_accleration
-        v1 = target_speed
-        a1 = target_acceleration
-        j0 = jerk_limit
-        j1 = -jerk_limit
-        delta_t = 0.001
-
-        velocity_from_start_time = []
-        velocity_from_end_time = []
+        delta_t = 0.01
+        vs_current = [current_speed]
+        vs_target = [target_speed]
+        as_current = [current_accleration]
+        as_target = [target_acceleration]
 
-        while v1 > v0:
-            velocity_from_start_time.append(v0)
-            velocity_from_end_time.append(v1)
-            v0 += a0 * delta_t
-            v1 -= a1 * delta_t
-            a0 = min(acceleration_limit, a0 + j0 * delta_t)
-            a1 = min(acceleration_limit, a1 - j1 * delta_t)
+        while vs_current[-1] < vs_target[-1]:
```

Comment:
I couldn't understand the purpose of re-iterating in a while loop. Please add some comments about the logic.

Reply:
Please check my answer to this question in #22, and continue any discussion there. This commit is only temporarily located in this pull-req until #22 is merged.